### ISSUE TITLE
Embeding LLamaSharp inside .NET MAUI Android App

### ISSUE THREAD

**Issue by Karimiprogramer** [SciSharp/LLamaSharp#954]
### Description

I am a .NET MAUI develpoer and I am interested in embeding a LLM inside my application with LLamaSharp. After build llama.cpp with ndk , How can I embed libraries on my application??

**Comment by andersonvieiragomeslopes** [comment]
> ### Description
> I am a .NET MAUI develpoer and I am interested in embeding a LLM inside my application with LLamaSharp. After build llama.cpp with ndk , How can I embed libraries on my application??

I'm also interested, did you manage to use it?

**Comment by Karimiprogramer** [comment]
After solving some problems with embeding model inside app , I will publish it into my github.

**Comment by Darlanio** [comment]
Tested on 0.19 without compiling the binaries myself and got the exception System.PlatformNotSupportedException: 'Operation is not supported on this platform.' on the ModelParams constructor.

**Comment by Darlanio** [comment]
Looking foward to Android supported nuget libraries.

**Comment by homerokzam** [comment]
Is there a solution for iOS?

**Comment by andersonvieiragomeslopes** [comment]
"Hey @AmSmart, could you join this discussion here? I saw that you had success with this [pull](#717 )."

**Comment by AmSmart** [comment]
Yeah, I have a PR that brings Android support for this library. Just that I haven't had enough time to write the tests and all. I'll open my PR this weekend and see if it's good enough to get merged. Also planning an IOS update in the future.

**Comment by amirvenus** [comment]
> Yeah, I have a PR that brings Android support for this library. Just that I haven't had enough time to write the tests and all. I'll open my PR this weekend and see if it's good enough to get merged. Also planning an IOS update in the future.

Any updates on this, @AmSmart ? 

This will be a really cool feature!

**Comment by Karimiprogramer** [comment]
I have successfully built CPU-based versions of libllama and libgguf for Android and imported them into my .NET MAUI application. However, I am facing some challenges in utilizing these libraries effectively within my app.

I would greatly appreciate any guidance or support you can provide to help me overcome these issues. Thank you!

**Comment by martindevans** [comment]
One thing to check - did you build _exactly_ the llama.cpp version listed in the table at the bottom of the readme? Each version of LLamaSharp is only compatible with one specific version of llama.cpp!

**Comment by AmSmart** [comment]
Update: The extra demand here gave me the motivation to complete my PR #1130 introducing Android support. Will still require a bit of back and forth but I hope it can get merged soon enough.

I've built this [nupkgs](https://1drv.ms/u/c/7c44b5ca50c7780b/Ecy6iKyyawNKkbvqQ0E3mRkBDNM_OcPkS_dawLMSAc1bNQ?e=KhowIE) off my PR so anyone blocked can use them to experiment temporarily. Reiterating that these are not official packages from the repo so it is definitely worth waiting for the next release with official Android support.

**Comment by alexdoit** [comment]
https://github.com/alexdoit/LLamaSharp-Android/releases/tag/v0.1-alpha
